crates/forge_main/src/config/handler.rs

```diff
@@ -0,0 +1,202 @@
+use std::str::FromStr;
+
+use anyhow::{Context, Result};
+use forge_api::{API, AgentId, ModelId, ProviderId};
+
+use super::display::{display_all_config, display_single_field, display_success};
+use super::error::{ConfigError, Result as ConfigResult};
+use crate::cli::{ConfigCommand, ConfigGetArgs, ConfigSetArgs};
+
+/// Handle config command
+pub async fn handle_config_command<A: API>(api: &A, command: ConfigCommand) -> Result<()> {
```

Comment:
convert to a struct `ConfigManager` that has access to Api. So we don't need to pipe api thru all the functions.